dashboard/app/models/course.rb

```diff
@@ -355,6 +355,16 @@ def has_older_version_progress?(user)
       count > 0
   end
 
+  # returns whether a script in this course has version_warning_dismissed.
+  def has_dismissed_version_warning?(user)
+    script_ids = default_scripts.map(&:id)
```

Comment:
nit: since [default_scripts](https://github.com/code-dot-org/code-dot-org/blob/e136b8a9659d1b4f6d64b20ee6e8f2f22d223696/dashboard/app/models/course.rb#L21) is an association (i.e. an `ActiveRecord::Relation`), we can do `default_script.pluck(:id)` (pluck instead of map) to only return ids from Sql rather than returning all columns and mapping in the client.